File: src/main/java/live/turna/phenyl/mirai/event/CBotOfflineEvent.java
package live.turna.phenyl.mirai.event;

import net.mamoe.mirai.event.events.BotOfflineEvent;
import net.md_5.bungee.api.plugin.Event;

/**
 * <b>CBotOfflineEvent</b><br>
 * BotOfflineEvent class.
 *
 * @since 2021/12/4 4:19
 */
public class CBotOfflineEvent extends Event {
    private final BotOfflineEvent event;
    private final String type;

    public CBotOfflineEvent(BotOfflineEvent event) {
        this.event = event;
        this.type = event.getClass().toString().split("\\$")[1];
    }

    /**
     * Get the type of dropping cause.<br>
     * Possible results:<br>
     * {@code Active}, {@code Force}, {@code MsfOffline}, {@code Dropped}, {@code RequireReconnect}.
     *
     * @return The type.
     * @see net.mamoe.mirai.event.events.BotOfflineEvent
     */
    public String getType() {
        return type;
    }

    /**
     * Get event bot QQ ID.
     *
     * @return The ID.
     */
    public Long getID() {
        return event.getBot().getId();
    }
}

File: src/main/java/live/turna/phenyl/mirai/event/CBotOfflineEvent.kt
package live.turna.phenyl.mirai.event

import net.mamoe.mirai.event.events.BotOfflineEvent
import net.md_5.bungee.api.plugin.Event

/**
 * **CBotOfflineEvent**<br></br>
 * BotOfflineEvent class.
 *
 * @since 2021/12/4 4:19
 */
class CBotOfflineEvent(event: BotOfflineEvent) : Event() {
    /**
     * Get the type of dropping cause.<br></br>
     * Possible results:<br></br>
     * `Active`, `Force`, `MsfOffline`, `Dropped`, `RequireReconnect`.
     *
     * @see net.mamoe.mirai.event.events.BotOfflineEvent
     */

    val type :String =event.javaClass.toString().split("\$").toTypedArray()[1]
    val id :Long = event.bot.id
}